### ISSUE TITLE
[REVIEW] api-security: add GraphQL DataLoader tenant-cache isolation checks

### ISSUE THREAD

**Issue by stmr** [UnitOneAI/SecuritySkills#2489]
# [REVIEW] api-security: add GraphQL DataLoader tenant-cache isolation checks

## Skill Being Reviewed
**Skill name:** `api-security`
**Skill path:** `skills/appsec/api-security/`

## False Positive Analysis

The skill correctly calls for resolver-level authorization, but a benign GraphQL DataLoader cache can look unsafe if it is scoped per request and includes the tenant boundary in the key.

Benign example:

```js
const loader = new DataLoader(ids => loadUsers(ctx.tenantId, ids), {
  cacheKeyFn: id => `${ctx.tenantId}:${id}`
});
```

This is not cross-tenant leakage if the loader is created per request and the cache key includes tenant or authorization scope.

## Coverage Gaps

Missed variant 1: a process-wide DataLoader caches `userId -> user` without tenant or permission scope, so tenant B can receive tenant A's object after a valid resolver lookup.

Missed variant 2: persisted GraphQL queries are allowlisted by hash, but authorization is only checked at query registration time and not at execution time.

Missed variant 3: subscription resolvers authorize the initial connection but do not re-check tenant access for each pushed event.

## Edge Cases

- Field masking can hide attributes in one resolver but leak the same object through a batched child resolver.
- Admin impersonation sessions need cache separation from the admin's normal user session.
- CDN/APQ caches may key by operation hash only, not by viewer scope.

## Remediation Quality

- Require request-scoped GraphQL loaders for sensitive data.
- Add cache key evidence: tenant, subject, role, and resource scope.
- Add resolver tests proving a warmed cache cannot cross tenant boundaries.

## Comparison to Other Tools

Semgrep and CodeQL can flag missing resolver authorization, but cross-request DataLoader cache scope is usually architectural. The skill should explicitly ask reviewers to inspect loader lifecycle and cache keys.

## Overall Assessment

Strong REST/GraphQL baseline. Add tenant-aware caching checks so BOLA/BFLA review covers GraphQL batching and persisted-query cache behavior.

## Bounty Info

- [x] I have read and agree to the `CONTRIBUTING.md` bounty terms.
- Preferred payment method: PayPal `samik4184@gmail.com`

**Comment by Errordog2** [comment]
/claim #2489

Opened implementation PR: https://github.com/UnitOneAI/SecuritySkills/pull/2515

Summary:
- Added tenant-scoped GraphQL batching/cache review gates for DataLoader lifecycle and cache-key scope.
- Documented the benign per-request DataLoader false-positive case from the issue.
- Added persisted query/APQ/CDN execution-time authorization checks.
- Added subscription event-level authorization checks and warmed-cache test expectations.

Validation:
- git diff --check
- Markdown fence balance check passed.
- Verified DataLoader, persisted-query/APQ/CDN, and subscription coverage markers in skills/appsec/api-security/SKILL.md.
